flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalogStoreFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.table.factories.CatalogStoreFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/** Catalog store factory for {@link GenericInMemoryCatalogStore}. */
+public class GenericInMemoryCatalogStoreFactory implements CatalogStoreFactory {
+    @Override
+    public CatalogStore createCatalogStore(Context context) {
+        return new GenericInMemoryCatalogStore();
+    }
+
+    @Override
+    public void open(Context context) {}
+
+    @Override
+    public void close() {}
+
+    @Override
+    public String factoryIdentifier() {
+        return GenericInMemoryCatalogStoreFactoryOptions.IDENTIFIER;
+    }
+
+    @Override
+    public Set<ConfigOption<?>> requiredOptions() {
+        return new HashSet<>();
+    }
+
+    @Override
+    public Set<ConfigOption<?>> optionalOptions() {
+        return new HashSet<>();
+    }
```

Comment:
Both of these options should not change after read, so IMO returning `Collections.emptySet()` would be better here. It is immutable and also spares the instantiation of a new object.